api.go

```diff
@@ -588,7 +587,15 @@ var successStatus = []int{
 // executeMethod - instantiates a given method, and retries the
 // request upon any error up to maxRetries attempts in a binomially
 // delayed manner using a standard back off algorithm.
-func (c Client) executeMethod(ctx context.Context, method string, metadata requestMetadata) (res *http.Response, err error) {
+func (c *Client) executeMethod(ctx context.Context, method string, metadata requestMetadata) (res *http.Response, err error) {
+	if c.IsOffline() {
+		return nil, &url.Error{
+			Op:  method,
+			URL: c.endpointURL.String(),
+			Err: errors.New("Connection closed by foreign host " + c.endpointURL.String() + ". Retry again."),
```

Comment:
Can we change this error message to avoid making it very standard ? Also why this is always foreign host that closed connection, it looks like c.IsOffline() can return true after networking connectivity.